### ISSUE TITLE
Prevent infinite logging loop with http

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/OrchardCore#2333]
When a user has an auth cookie created in https, a new session using http won't be able to authenticate. This is due to the fact that the server sends back the auth cookie in http, but the browser rejects it because it already has one in https. And the the one in https is not sent because the navigation is done in http.

**Comment by bleroy** [comment]
Just want to point out that displaying a login screen at all over HTTP is not something that should be done, ever. We all know that of course, but seems relevant to how we choose to fix this bug...

**Comment by sebastienros** [comment]
Then all https configuration (including any https related feature) should be done by code or configuration, and not from an admin screen (like today)

**Comment by bleroy** [comment]
Yup, I agree. Although at this point, what configuration remains to be done beyond setting up the cert?

**Comment by sebastienros** [comment]
- Forcing HTTPS for the whole site (redirections)
- Permanent redirections
- Enabling HSTS

**Comment by bleroy** [comment]
Right, so there's a small number of parameters there (ports, expiration, etc.) with reasonable defaults. I'm suggesting that the fact that HTTPS is forced is not even a setting, it's always true. If you have a cert set-up already, it's not out of the question that everything could be setup to just work.

**Comment by Piedone** [comment]
Nowadays this needs to be a non-issue: If you access a site via HTTPS, then (if competently configured) you'll never access it via HTTP. Is this still applicable?
